### PR TITLE
Feat/report token and failures

### PR DESCRIPTION
**Pull request by powerje**
Adds optional delegate methods for getting the push token and any errors that occur while registering for push tokens, per discussion in https://github.com/Azure/azure-notificationhubs-ios/issues/43

### CODE REVIEW

File: src/WindowsAzureMessaging/WindowsAzureMessaging/include/WindowsAzureMessaging/MSNotificationHubDelegate.h [hunk 1]
```diff
@@ -18,9 +18,29 @@
...
     didReceivePushNotification:(MSNotificationHubMessage *)message
         fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler;
 
+/**
+ * Callback method that will be called whenever the push token is retrieved.
+ *
+ * @param notificationHub The instance of MSNotificationHub
+ * @param pushToken The push token
+ */
+- (void)notificationHub:(MSNotificationHub *)notificationHub
```
Comment (mpodwysocki) [inline src/WindowsAzureMessaging/WindowsAzureMessaging/include/WindowsAzureMessaging/MSNotificationHubDelegate.h:33]: If we're going to do a rename, we should have maybe a sample linked in our Swift sample app?

**Comment by powerje** [comment]
@mpodwysocki apologies on not getting to the requested changes yet. Let me know if there's any urgency on Microsoft's end to get these in sooner than later. I will be able to allocate time for this around the 22nd based on my sprint schedule but could move it up in priority if need be to hit a Microsoft release target.

**Comment by mpodwysocki** [comment]
@powerje We're looking to GA fairly soon, although this is not a breaking change at this point, but an addition, so it can certainly be available after that.  We're also adding push to user and other features so there are opportunities forthcoming.

**Comment by mpodwysocki** [comment]
> @mpodwysocki apologies on not getting to the requested changes yet. Let me know if there's any urgency on Microsoft's end to get these in sooner than later. I will be able to allocate time for this around the 22nd based on my sprint schedule but could move it up in priority if need be to hit a Microsoft release target.

As you can tell, we've made a lot of changes since your initial PR.  Can you update it to reflect the newest changes?

**Pull request closed without merging**